data/blog/advancements-in-lightning-infrastructure.mdx

```diff
@@ -0,0 +1,374 @@
+As the Lightning Network evolves, the need for modular, open-source node
+implementations has become increasingly clear. Existing implementations often
+come with trade-offs that limit flexibility for developers seeking highly
+customizable solutions. Addressing this gap, [Lampo](https://lampo.devcrew.cc/)
+provides a modular, community-driven alternative, built on the Lightning
+Development Kit (LDK).
+
+With [support](/blog/8th-wave-of-bitcoin-grants#lampo) from OpenSats, the
+project has achieved a significant milestone by contributing to the integration
+of LDK with `lnprototest`, a test suite designed to rigorously validate
+Lightning protocol behavior. This work enhances LDK's reliability, ensures
+adherence to Lightning standards, and strengthens its overall robustness, giving
+developers greater confidence when testing their implementations. A key part of
+this effort was the launch of the
+[`ldk-lnprototest`](https://pypi.org/project/ldk-lnprototest/) package on PyPI.
+This package provides a runner that facilitates the integration of LDK with
+`lnprototest`, enabling automated and standardized testing. By reducing the
+risks of protocol inconsistencies and implementation errors—challenges that can
+significantly impact Lightning's reliability—this tool ensures a more robust and
+predictable development process for LDK-based implementations like Lampo.
+
+>Building reliable Lightning infrastructure requires both rigorous testing and
+>continuous improvements. The LDK-Lnprototest integration and enhancements to
+>Lampo aim to make the network more robust and developer-friendly. Excited to
+>push this forward! 
+>—Prakhar Saxena, Lampo Project Lead
+
+In addition to improving the Lightning testing environment, Lampo has also
+[introduced
+improvements](https://github.com/vincenzopalazzo/lampo.rs/pull/295/commits) to
+wallet security and usability. Previously, users had to manually enter their
+[BIP39](https://github.com/bitcoin/bips/blob/master/bip-0039.mediawiki) seed words in the
+command line every time they restored a wallet, posing privacy risks and
+reducing usability. Now, Lampo securely stores and manages wallet restoration
+data in a binary file (`wallet.dat`), eliminating the need for repeated manual
+entry and improving the user experience without compromising security.
+
+Together, these improvements make Lampo both a reliable development tool and a
+more secure node implementation.
+
+Looking ahead, Lampo plans to continue its evolution by implementing a
+proof-of-concept for Validating Lightning Signer (VLS), a move that will further
+improve security by enabling external validation of Lightning signatures. Future
+updates will also introduce
+[BOLT12](https://github.com/lightning/bolts/blob/master/12-offer-encoding.md#bolt-12-negotiation-protocol-for-lightning-payments)
+support, refine the plugin system, and expand documentation to encourage broader
+adoption. By fostering a truly open-source, modular approach to running a
```

Comment:
```suggestion
Looking ahead, Lampo plans to continue its evolution by implementing a
proof-of-concept for Validating Lightning Signer (VLS), a move that will further
improve security by enabling external validation of Lightning signatures. Future
updates will also introduce a PoC of a plugin system, and expand documentation to encourage broader
adoption. By fostering a truly open-source, modular approach to running a
```

Lampo supported BOLT 12 before ocean released their support, you can see lampo as one of the first lightning network implementations that support the recent version of bolt12 thanks to ldk